tests/bgp/test_startup_tsa_tsb_service.py

```diff
@@ -409,15 +439,17 @@ def test_tsa_tsb_service_with_supervisor_cold_reboot(duthosts, localhost, enum_s
     suphost = duthosts[enum_supervisor_dut_hostname]
     tsa_tsb_timer = dict()
     dut_nbrhosts = dict()
+    up_bgp_neighbors = dict()
     orig_v4_routes, orig_v6_routes = dict(), dict()
     for linecard in duthosts.frontend_nodes:
+        up_bgp_neighbors[linecard] = linecard.get_bgp_neighbors_per_asic("established")
         tsa_tsb_timer[linecard] = get_startup_tsb_timer(linecard)
         if not tsa_tsb_timer[linecard]:
             pytest.skip("startup_tsa_tsb.service is not supported on the duts under {}".format(suphost.hostname))
         dut_nbrhosts[linecard] = nbrhosts_to_dut(linecard, nbrhosts)
         # Ensure that the DUT is not in maintenance already before start of the test
-        pytest_assert(TS_NORMAL == get_traffic_shift_state(linecard),
-                      "DUT is not in normal state")
+        if TS_NORMAL != get_traffic_shift_state(linecard):
+            pytest.skip("Test skipped since DUT is not in normal state")
```

Comment:
if the previous tests fail and leave the testbed in bad state, I think we should not skip the test. The assert here is correct

Reply:
Hi @arlakshm - Explained the reasoning in the above comment